src/queries/a11yStates.ts

```diff
@@ -0,0 +1,79 @@
+import type { ReactTestInstance } from 'react-test-renderer';
+import { AccessibilityState } from 'react-native';
+import { makeQueries } from './makeQueries';
+import type {
+  FindAllByQuery,
+  FindByQuery,
+  GetAllByQuery,
+  GetByQuery,
+  QueryAllByQuery,
+  QueryByQuery,
+} from './makeQueries';
+
+type AccessibilityStateKey = keyof AccessibilityState;
+type AccessibilityStateKeyQueryParam =
+  | AccessibilityStateKey
+  | Array<AccessibilityStateKey>;
+
+export function matchArrayValue(
+  prop: Array<string> | undefined,
+  matcher: string | Array<string>
+): boolean {
+  if (!prop || matcher.length === 0) {
+    return false;
+  }
+
+  if (typeof matcher === 'string') {
+    return prop.includes(matcher);
+  }
+
+  return !matcher.some((e) => !prop.includes(e));
+}
+
+const queryAllByA11yStates = (
+  instance: ReactTestInstance
+): ((
+  accessibilityStateKey: AccessibilityStateKeyQueryParam
+) => Array<ReactTestInstance>) =>
+  function queryAllByDisplayValueFn(accessibilityStateKey) {
+    return instance.findAll(
+      (node) =>
+        typeof node.type === 'string' &&
+        matchArrayValue(node.props.accessibilityState, accessibilityStateKey)
+    );
+  };
+
+const getMultipleError = (a11yStates: AccessibilityStateKeyQueryParam) =>
+  `Found multiple elements with accessibilityState: ${JSON.stringify(
+    a11yStates
+  )}`;
+const getMissingError = (a11yStates: AccessibilityStateKeyQueryParam) =>
+  `Unable to find an element with accessibilityState: ${JSON.stringify(
+    a11yStates
+  )}`;
+
+const { getBy, getAllBy, queryBy, queryAllBy, findBy, findAllBy } = makeQueries(
+  queryAllByA11yStates,
+  getMissingError,
+  getMultipleError
+);
+
+export type ByA11yStatesQueries = {
+  getByA11yStates: GetByQuery<AccessibilityStateKeyQueryParam>;
```

Comment:
Add alias names `getByAccessiblityStates`, etc. See comments in `a11yHint.ts` for details.